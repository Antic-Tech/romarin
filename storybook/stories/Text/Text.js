import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import theme from '../../theme';

const RomarinText = props => {
	const fontSizes = {
		xlarge: theme.fontSizeXLarge,
		large: theme.fontSizeLarge,
		medium: theme.fontSizeMedium,
		small: theme.fontSizeSmall,
		xsmall: theme.fontSizeXSmall
	};

	return (
		<Text
			style={{
				fontSize: fontSizes[props.size],
				color: props.muted ? theme.textColorMuted : theme.textColor
			}}
		>
			{props.children}
		</Text>
	);
};

export { RomarinText as default };
