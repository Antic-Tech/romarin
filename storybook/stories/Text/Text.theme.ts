import { BaseTheme } from '../../definitions/Base.theme.d';
import { TextTheme } from '../../definitions/Text.theme.d';

function getTextTheme(core:BaseTheme): TextTheme{
	return {
		color: core.colorBlack,
		colorMuted: core.colorLightGray,
		colorTouchable: core.colorPrimary,
	}
};

export default getTextTheme;
