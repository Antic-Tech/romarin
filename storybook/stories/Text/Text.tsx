import PropTypes from 'prop-types';
import React from 'react';
import { Text } from 'react-native';
import { withTheme } from '../../theme';

const RomarinText = (props) => {
	const fontSizes = {
		xlarge: props.theme!.base!.fontSizeXLarge,
		large: props.theme!.base!.fontSizeLarge,
		medium: props.theme!.base!.fontSizeMedium,
		small: props.theme!.base!.fontSizeSmall,
		xsmall: props.theme!.base!.fontSizeXSmall
	};
	const { size, muted, color, weight, children, style = {} } = props;
	const textColor = color ? color : muted ? props.theme.text.colorMuted : props.theme.text.color;
	return (
		<Text
			style={[
				{
					fontSize: fontSizes[size],
					fontFamily: props.theme.fontFamily,
					color: textColor,
					fontWeight: weight
				},
				style
			]}
		>
			{children}
		</Text>
	);
};

RomarinText.propTypes = {
	size: PropTypes.string,
	muted: PropTypes.bool
};

RomarinText.defaultProps = {
	size: 'medium',
	muted: false
};

export default withTheme(RomarinText);
