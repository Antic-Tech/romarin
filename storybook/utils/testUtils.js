import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

export const renderComponent = (jsx, { shallowRender = true, store } = {}) => {
	const renderOptions = { context: { store } };

	if (shallowRender) {
		return shallow(jsx, renderOptions);
	}

	return mount(jsx, renderOptions);
};

export const componentFactory = Component => (props = {}, options = {}) => {
	return renderComponent(<Component {...props} />, options);
};

export const getTree = jsx => renderer.create(jsx).toJSON();

export const verifySnapshot = jsx => {
	const tree = getTree(jsx);

	expect(tree).toMatchSnapshot();
};
