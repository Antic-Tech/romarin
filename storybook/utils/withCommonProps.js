import React from 'react';
import { StyleSheet } from 'react-native';
import ThemeProvider from '../theme';

const Sizes = {
	XXSmall: 'XXSmall',
	XSmall: 'XSmall',
	Small: 'Small',
	Medium: 'Medium',
	Large: 'Large',
	XLarge: 'XLarge',
	XXLarge: 'XXLarge'
};

const withPadding = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				padding,
				paddingHorizontal,
				paddingVertical,
				paddingTop,
				paddingBottom,
				paddingLeft,
				paddingRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					padding: padding ? ThemeProvider.theme[`corePadding${padding}`] : undefined,
					paddingHorizontal: paddingHorizontal
						? ThemeProvider.theme[`corePadding${paddingHorizontal}`]
						: undefined,
					paddingVertical: paddingVertical ? ThemeProvider.theme[`corePadding${paddingVertical}`] : undefined,
					paddingTop: paddingTop ? ThemeProvider.theme[`corePadding${paddingTop}`] : undefined,
					paddingBottom: paddingBottom ? ThemeProvider.theme[`corePadding${paddingBottom}`] : undefined,
					paddingLeft: paddingLeft ? ThemeProvider.theme[`corePadding${paddingLeft}`] : undefined,
					paddingRight: paddingRight ? ThemeProvider.theme[`corePadding${paddingRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withMargin = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				margin,
				marginHorizontal,
				marginVertical,
				marginTop,
				marginBottom,
				marginLeft,
				marginRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					margin: margin ? ThemeProvider.theme[`coreMargin${margin}`] : undefined,
					marginHorizontal: marginHorizontal
						? ThemeProvider.theme[`coreMargin${marginHorizontal}`]
						: undefined,
					marginVertical: marginVertical ? ThemeProvider.theme[`coreMargin${marginVertical}`] : undefined,
					marginTop: marginTop ? ThemeProvider.theme[`coreMargin${marginTop}`] : undefined,
					marginBottom: marginBottom ? ThemeProvider.theme[`coreMargin${marginBottom}`] : undefined,
					marginLeft: marginLeft ? ThemeProvider.theme[`coreMargin${marginLeft}`] : undefined,
					marginRight: marginRight ? ThemeProvider.theme[`coreMargin${marginRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withCommonProps = (Component) => withPadding(withMargin(Component));

export { withPadding, withMargin, withCommonProps };
