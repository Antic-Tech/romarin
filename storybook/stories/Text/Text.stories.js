import React from "react";

import { storiesOf } from "@storybook/react-native";
import { action } from "@storybook/addon-actions";

import Text from "./Text";
import CenterView from "../CenterView";

storiesOf("Text", module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add("with text", () => (
    <React.Fragment>
      <Text size="xlarge">xlarge</Text>
      <Text size="large">large</Text>
      <Text size="medium">medium</Text>
      <Text size="small">small</Text>
      <Text size="xsmall">xsmall</Text>

      <Text muted>medium</Text>
    </React.Fragment>
  ));
