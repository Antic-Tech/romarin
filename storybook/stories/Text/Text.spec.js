import React from 'react';

import Text from './Text';

import { verifySnapshot } from '../../utils/testUtils';

describe('Text', () => {
	describe('Snapshots', () => {
		it('renders correctly', () => {
			verifySnapshot(<Text>Text</Text>);
		});
		it('renders xlarge correctly', () => {
			verifySnapshot(<Text size="xlarge">Text</Text>);
		});
		it('renders large correctly', () => {
			verifySnapshot(<Text size="large">Text</Text>);
		});
		it('renders medium correctly', () => {
			verifySnapshot(<Text size="medium">Text</Text>);
		});
		it('renders small correctly', () => {
			verifySnapshot(<Text size="small">Text</Text>);
		});
		it('renders small correctly', () => {
			verifySnapshot(<Text xsize="small">Text</Text>);
		});
		it('renders muted correctly', () => {
			verifySnapshot(<Text muted>Text</Text>);
		});
	});
});
