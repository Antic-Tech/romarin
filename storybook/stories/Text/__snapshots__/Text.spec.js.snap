// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Snapshots renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#333333",
      "fontSize": undefined,
    }
  }
>
  Text
</Text>
`;

exports[`Text Snapshots renders large correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#333333",
      "fontSize": 28,
    }
  }
>
  Text
</Text>
`;

exports[`Text Snapshots renders medium correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#333333",
      "fontSize": 17,
    }
  }
>
  Text
</Text>
`;

exports[`Text Snapshots renders muted correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#999999",
      "fontSize": undefined,
    }
  }
>
  Text
</Text>
`;

exports[`Text Snapshots renders small correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#333333",
      "fontSize": 13,
    }
  }
>
  Text
</Text>
`;

exports[`Text Snapshots renders small correctly 2`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#333333",
      "fontSize": undefined,
    }
  }
>
  Text
</Text>
`;

exports[`Text Snapshots renders xlarge correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#333333",
      "fontSize": 34,
    }
  }
>
  Text
</Text>
`;
